### PR TITLE
Managed identity support for device-identity import and export 

### PR DESCRIPTION
**Pull request by c-ryan-k**
Pending merge and release of 
https://github.com/Azure/azure-cli/pull/18098

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [ ] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

File: azext_iot/tests/iothub/test_iot_ext_int.py [hunk 1]
```diff
@@ -148,7 +228,110 @@ def test_identity_storage(self):
...
+                self.check("failureReason", None),
+                self.check("type", "export"),
+                self.check("excludeKeysInExport", False),
+                self.check("storageAuthenticationType", "identityBased"),
+                self.exists("jobId"),
+            ],
+        )
+
+        # give time to finish job
+        from time import sleep
```
Comment (digimaun) [inline azext_iot/tests/iothub/test_iot_ext_int.py:301]: The import for sleep is happening so many times might as well import once at the top of the module.

File: azext_iot/tests/iothub/test_iot_ext_int.py [hunk 2]
```diff
@@ -32,10 +33,28 @@
 
 CWD = os.path.dirname(os.path.abspath(__file__))
 
+test_managed_name = "TESTMANAGEDIDENTITY"
```
Comment (digimaun) [inline azext_iot/tests/iothub/test_iot_ext_int.py:36]: Could generate a unique name ? We use `azext_iot.tests.generators.generate_generic_id` across tests for this purpose.

File: azext_iot/tests/iothub/test_iot_ext_int.py [hunk 3]
```diff
@@ -32,10 +33,28 @@
...
+        self.managed_identity = None
+
+    def get_managed_identity(self):
+        # Check if there is a managed identity already
+        if self.managed_identity:
+            return self.managed_identity
+
+        # Create managed identity
+        result = self.cmd(
+            "identity create -n {} -g {}".format(
```
Comment (digimaun) [inline azext_iot/tests/iothub/test_iot_ext_int.py:51]: Is this a create or update, or an idempotent operation i.e. what happens if an identity with the same name exists

File: azext_iot/tests/iothub/test_iot_ext_int.py (hunk 2 above)
Comment (digimaun) [inline azext_iot/tests/iothub/test_iot_ext_int.py:36]: Would be good to slightly tweak the current `LIVE_STORAGE` and `LIVE_STORAGE_ID` variables to be more explicit/reduce reliance on comment. Maybe `LIVE_STORAGE` -> `LIVE_STORAGE_URI` || `LIVE_STORAGE_SAS_URI`  and `LIVE_STORAGE_ID` -> `LIVE_STORAGE_RESOURCE_ID`.

File: azext_iot/tests/iothub/test_iot_ext_int.py (hunk 3 above)
Comment (digimaun) [reply azext_iot/tests/iothub/test_iot_ext_int.py:51]: We should consider clean up of the identity after its usage in the test as well.

**Comment by c-ryan-k** [comment]
image
Had a bit of a hiccup in user-identity timing, but seems to be relatively stable.

**Review by digimaun — APPROVED** [review]

**Pull request merged**